modules/yieldbotBidAdapter.js

```diff
@@ -0,0 +1,614 @@
+   * A session may be blocked for efficiency (i.e. Yieldbot has decided no to bid for the session),
+   * security and/or fraud detection.
+   * @returns {boolean}
+   * @readonly
+   * @memberof module:YieldbotBidAdapter
+   * @private
+   */
+  get sessionBlocked() {
+    const cookieName = '__ybotn';
+    const cookieValue = this.getCookie(cookieName);
+    const sessionBlocked = cookieValue ? parseInt(cookieValue, 10) || 0 : 0;
+    if (sessionBlocked) {
+      this.setCookie(cookieName, 1, this._sessionTimeout, '/');
+    }
+    this._sessionBlocked = !!sessionBlocked;
+    return this._sessionBlocked;
+  },
+
+  set sessionBlocked(blockSession) {
+    const cookieName = '__ybotn';
+    const sessionBlocked = blockSession ? 1 : 0;
+    this.setCookie(cookieName, sessionBlocked, this._sessionTimeout, '/');
+    return !!sessionBlocked;
+  },
+
+  get userId() {
+    const cookieName = '__ybotu';
+    let cookieValue = this.getCookie(cookieName);
+    if (!cookieValue) {
+      cookieValue = this.newId();
+      this.setCookie(cookieName, cookieValue, this._userTimeout, '/');
+    }
+    return cookieValue;
+  },
+
+  get sessionId() {
+    const cookieName = '__ybotsi';
+    let cookieValue = this.getCookie(cookieName);
+    if (!cookieValue) {
+      cookieValue = this.newId();
+      this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
+    }
+    return cookieValue;
+  },
+
+  get pageviewDepth() {
+    const cookieName = '__ybotpvd';
+    let cookieValue = parseInt(this.getCookie(cookieName), 10) || 0;
+    cookieValue++;
+    this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
+    return cookieValue;
+  },
+
+  get lastPageviewId() {
+    const cookieName = '__ybotlpvi';
+    let cookieValue = this.getCookie(cookieName);
+    return cookieValue || '';
+  },
+
+  set lastPageviewId(id) {
+    const cookieName = '__ybotlpvi';
+    return this.setCookie(cookieName, id, this._sessionTimeout, '/');
+  },
+
+  get lastPageviewTime() {
+    const cookieName = '__ybotlpv';
+    let cookieValue = this.getCookie(cookieName);
+    return cookieValue ? parseInt(cookieValue, 10) : 0;
+  },
+
+  set lastPageviewTime(ts) {
+    const cookieName = '__ybotlpv';
+    return this.setCookie(cookieName, ts, this._sessionTimeout, '/');
+  },
+
+  /**
+   * Get/set the request base url used to form bid, ad markup and impression requests.
+   * @param {string} [prefix] the bidder request base url
+   * @returns {string} the request base Url string
+   * @memberof module:YieldbotBidAdapter
+   */
+  urlPrefix: function(prefix) {
+    const cookieName = '__ybotc';
+    const pIdx = prefix ? prefix.indexOf(':') : -1;
+    const url = pIdx !== -1 ? document.location.protocol + prefix.substr(pIdx + 1) : null;
+    let cookieValue = url || this.getCookie(cookieName);
+    if (!cookieValue) {
+      cookieValue = '//i.yldbt.com/m/';
+    }
+    this.setCookie(cookieName, cookieValue, this._sessionTimeout, '/');
```

Comment:
why do you set this into a cookie?

Reply:
The Yieldbot adserver response will always include a property, `url_prefix`, that specifies the appropriate availability zone domain for bid requests. e.g.

`{url_prefix: 'http://ads-adseast-vpc.yldbt.com/m/'}`

https://github.com/prebid/Prebid.js/pull/2135/files#diff-0bfb815458bc7804a5ccc404bcf829d2R181